src/deriveNativeScriptHash.c

```diff
@@ -169,6 +173,10 @@ static void deriveScriptHash_display_ui_runStep()
 			// max possible length 85: "Requires n out of k signatures. Contains k nested scripts."
 			// where n and k is 2^32-1
 			char text[86] = {0};
+			STATIC_ASSERT(sizeof(unsigned) >= sizeof(ctx->scriptContent.requiredScripts), "oversized type for %u");
+			STATIC_ASSERT(sizeof(unsigned) >= sizeof(ctx->complexScripts[ctx->level].remainingScripts), "oversized type for %u");
+			STATIC_ASSERT(!IS_SIGNED(ctx->scriptContent.requiredScripts), "signed type for %u");
```

Comment:
I guess I'd prefer to group these by the variable that is being checked, not the check type, but don't feel strongly about it.

Reply:
Also, it might be worth to have local variables to avoid repeating `ctx->complexScripts[ctx->level].remainingScripts` and `ctx->scriptContent.requiredScripts` again and again.

Reply:
Can be done, it's a taste thing, I don't feel strongly about it either.

I like to repeat the expressions, because it shows that we check the exact same thing, without obfuscating the original code. And it's compile time only.

Reply:
OK, let's keep the order of checks as it is, just get rid of the long variable references.